osbenchmark/workload/params.py

```diff
@@ -885,22 +886,53 @@ def __init__(self, workload, params, context: Context, **kwargs):
         self.field_name: str = parse_string_parameter("field", params)
         self.context = context
         self.data_set_format = parse_string_parameter("data_set_format", params)
-        self.data_set_path = parse_string_parameter("data_set_path", params)
+        self.data_set_path = parse_string_parameter("data_set_path", params, "")
+        self.data_set_corpus = parse_string_parameter("data_set_corpus", params, "")
+        self._validate_data_set(self.data_set_path, self.data_set_corpus)
         self.total_num_vectors: int = parse_int_parameter("num_vectors", params, -1)
         self.num_vectors = 0
         self.total = 1
         self.current = 0
         self.percent_completed = 0
         self.offset = 0
         self.data_set: DataSet = None
+        self.corpora = self.extract_corpora(self.data_set_corpus, self.data_set_format)
+
+    def _get_corpora_file_paths(self, name, source_format):
+        document_files = []
+        for corpus in self.corpora:
+            if corpus.name != name:
+                continue
+            filtered_corpus = corpus.filter(source_format=source_format)
+            document_files.extend([document.document_file for document in filtered_corpus.documents])
+        return document_files
 
     @property
     def infinite(self):
         return False
 
-    def _is_last_partition(self, partition_index, total_partitions):
+    @staticmethod
+    def _is_last_partition(partition_index, total_partitions):
         return partition_index == total_partitions - 1
 
+    @staticmethod
+    def _validate_data_set(file_path, corpus):
+        if not file_path and not corpus:
+            raise exceptions.ConfigurationError(
+                "Dataset is missing. Provide either dataset file path or valid corpus.")
+        if file_path and corpus:
+            raise exceptions.ConfigurationError(
+                "Provide either dataset file path '%s' or corpus '%s'." % (file_path, corpus))
```

Comment:
For more clarity, let's add more details in the exception:
```
"User provided both file path and corpus. Provide either dataset file path '%s' or corpus '%s', but not both" % (file_path, corpus))
```

Reply:
Updated